packages/flutter/lib/src/material/selectable_text.dart

```diff
@@ -755,51 +755,57 @@ class _SelectableTextState extends State<SelectableText>
-        cursorRadius: cursorRadius,
-        cursorColor: cursorColor,
-        selectionHeightStyle: widget.selectionHeightStyle,
-        selectionWidthStyle: widget.selectionWidthStyle,
-        cursorOpacityAnimates: cursorOpacityAnimates,
-        cursorOffset: cursorOffset,
-        paintCursorAboveText: paintCursorAboveText,
-        backgroundCursorColor: CupertinoColors.inactiveGray,
-        enableInteractiveSelection: widget.enableInteractiveSelection,
-        magnifierConfiguration:
-            widget.magnifierConfiguration ?? TextMagnifier.adaptiveMagnifierConfiguration,
-        dragStartBehavior: widget.dragStartBehavior,
-        scrollPhysics: widget.scrollPhysics,
-        scrollBehavior: widget.scrollBehavior,
-        autofillHints: null,
-        contextMenuBuilder: widget.contextMenuBuilder,
-      ),
+    Widget editable = EditableText(
+      key: editableTextKey,
+      style: effectiveTextStyle,
+      readOnly: true,
+      toolbarOptions: widget.toolbarOptions,
+      textWidthBasis: widget.textWidthBasis ?? defaultTextStyle.textWidthBasis,
+      textHeightBehavior: widget.textHeightBehavior ?? defaultTextStyle.textHeightBehavior,
+      showSelectionHandles: _showSelectionHandles,
+      showCursor: widget.showCursor,
+      controller: _controller,
+      focusNode: focusNode,
+      strutStyle: widget.strutStyle ?? const StrutStyle(),
+      textAlign: widget.textAlign ?? defaultTextStyle.textAlign ?? TextAlign.start,
+      textDirection: widget.textDirection,
+      textScaler: effectiveScaler,
+      autofocus: widget.autofocus,
+      forceLine: false,
+      minLines: widget.minLines,
+      maxLines: widget.maxLines ?? defaultTextStyle.maxLines,
+      selectionColor: widget.selectionColor ?? selectionColor,
+      selectionControls: widget.selectionEnabled ? textSelectionControls : null,
+      onSelectionChanged: _handleSelectionChanged,
+      onSelectionHandleTapped: _handleSelectionHandleTapped,
+      rendererIgnoresPointer: true,
+      cursorWidth: widget.cursorWidth,
+      cursorHeight: widget.cursorHeight,
+      cursorRadius: cursorRadius,
+      cursorColor: cursorColor,
+      selectionHeightStyle: widget.selectionHeightStyle,
+      selectionWidthStyle: widget.selectionWidthStyle,
+      cursorOpacityAnimates: cursorOpacityAnimates,
+      cursorOffset: cursorOffset,
+      paintCursorAboveText: paintCursorAboveText,
+      backgroundCursorColor: CupertinoColors.inactiveGray,
+      enableInteractiveSelection: widget.enableInteractiveSelection,
+      magnifierConfiguration:
+          widget.magnifierConfiguration ?? TextMagnifier.adaptiveMagnifierConfiguration,
+      dragStartBehavior: widget.dragStartBehavior,
+      scrollPhysics: widget.scrollPhysics,
+      scrollBehavior: widget.scrollBehavior,
+      autofillHints: null,
+      contextMenuBuilder: widget.contextMenuBuilder,
     );
+    final ui.TypographySettings? typographySettings = MediaQuery.maybeTypographySettingsOf(context);
```

Comment:
If I'm an app developer and my `Text` or `TextField` widget has unexpected styling, I might find it tricky to understand how `DefaultTextStyle` and `MediaQuery` interact. It might be worth expanding docs a little:

1. `DefaultTextStyle`'s docs should perhaps explain that `MediaQuery` can override text styling properties.
2. `Text.style`'s docs should perhaps explain that `MediaQuery` can override text styling properties.
3. `TextField.style` docs should perhaps explain that `MediaQuery` can override text styling properties.

Reply:
Good catch! Will definitely add docs around these overrides.